redhat-cop/automate_tower_genie_collections
# Reduce memory utilization when exporting objects and organize the output files in directories

- Reduce the number of used variables.
- Organize the output in directories.
- Add the object ID as part of the generated file

### What does this PR do?
When exporting `job_templates` (and can be extensible to other objects), we get out of resources:
![image](https://user-images.githubusercontent.com/26822043/190576560-127d4499-2848-4d23-88af-98653f061178.png)
```
...
ok: [localhost] => (item=ORGANIZATIONLESS - TestLoop)
ok: [localhost] => (item=ORGANIZATIONLESS - Reboot Servers - [PROD])
ok: [localhost] => (item=ORGANIZATIONLESS - Reboot Servers - [NONPROD])
ERROR! A worker was found in a dead state

real    16m0.070s
user    13m23.096s
sys     1m45.619s
(venv) [lab] $ 
```
This PR is fixing this issue (it appears in large environments -more than 300 Job Templates, inventories, etc.- and organizing the output better.

### How should this be tested?
```
$ cat filetree_create.yml
---
- hosts: localhost
  connection: local
  gather_facts: false
  vars:
    controller_hostname: "{{ vault_controller_hostname }}"
    controller_username: "{{ vault_controller_username }}"
    controller_password: "{{ vault_controller_password }}"
    controller_validate_certs: "{{ vault_controller_validate_certs }}"
    controller_api_plugin: "ansible.controller.controller_api"
  pre_tasks:
    - name: "Check if the required input variables are present"
      assert:
        that:
          - input_tag is defined
          - (input_tag | type_debug) == 'list'
        fail_msg: 'A variable called ''input_tag'' of type ''list'' is needed: -e ''{input_tag: [organizations, projects]}'''
        quiet: true

    - name: "Check if the required input values are correct"
      assert:
        that:
          - tag_item in valid_tags
        fail_msg: "The valid tags are the following ones: {{ valid_tags | join(', ') }}"
        quiet: true
      loop: "{{ input_tag }}"
      loop_control:
        loop_var: tag_item
  roles:
    - redhat_cop.controller_configuration.filetree_create
...
$ ansible-playbook filetree_create.yml -e '{input_tag: [job_templates], output_path: ~/iam/filetree_create.2.1.7 }'
```
### Is there a relevant Issue open for this?
No issue opened for this.

### Other Relevant info, PRs, etc.
N/A
